app/assets/javascripts/discourse/app/models/login-method.js

```diff
@@ -67,6 +68,13 @@ export default class LoginMethod extends EmberObject {
       params["signup"] = true;
     }
 
+    const destinationUrl = cookie("destination_url");
```

Comment:
When using an "external login" method (aka. social logins), we pass the "origin" to the server (omniauth).